Modules/IO/TIFF/src/itkTIFFImageIO.cxx

```diff
@@ -974,6 +1000,52 @@ TIFFImageIO::PopulateColorPalette()
   }
 }
 
+void
+TIFFImageIO::AllocateTiffPalette(int bps)
+{
+  m_ColorRed = nullptr;
+  m_ColorGreen = nullptr;
+  m_ColorBlue = nullptr;
+
+  uint16 bpp = static_cast<uint16>(bps);
+  m_ColorRed = static_cast<uint16 *>(_TIFFmalloc(1 << bpp * sizeof(uint16)));
```

Comment:
`D:\a\1\s\Modules\IO\TIFF\src\itkTIFFImageIO.cxx(1011): warning C4334: '<<': result of 32-bit shift implicitly converted to 64 bits (was 64-bit shift intended?)` and a [few more times](https://open.cdash.org/viewBuildError.php?type=1&buildid=6087738).

Reply:
I think I've stolen  this code from [here](https://www.asmail.be/msg0055228619.html). [Another](http://maptools-org.996276.n3.nabble.com/Adding-alpha-to-palette-files-td13203.html) example.  I think here, it is mainly used for computing fast power of 2 `1 << (bpp * sizeof(uint16)) == std::pow(2, (bpp * sizeof(uint16))) `.  So 64 or 32 bit shift would give the same result. I could just static cast the result to a size t (input type of TIFFmalloc).

Reply:
> computing fast power of 2

That is clear. But let's get rid of the warning in the least intrusive way. Maybe make `bpp` variable `uint8`?

Reply:
Would uint8 it make a difference ? What about :
```
  tmsize_t array_size = 1 << bpp * sizeof(uint16);
  m_ColorRed = static_cast<uint16 *>(_TIFFmalloc(array_size));
  if (!m_ColorRed)
  {
    _TIFFfree(m_ColorRed);
    itkExceptionMacro("Can't allocate space for Red channel of component tables.");
  }
```
It is anyway computed 3 times. I don't see any warning when I compile on my machine. Do you have any advice to see those ?

Reply:
When configuring ITK, CMAKE_CXX_FLAGS is set to `/DWIN32 /D_WINDOWS /W3 /GR /EHsc` by default. Change `/W3` into `/W4` to increase warning level.